### PR TITLE
OADP-3954 ImageStreamTag Update

### PR DESCRIPTION
**Pull request by shdeshpa07**
## Jira 

* [OADP-3954](https://issues.redhat.com/browse/OADP-3954)

Added a note to address the issue and the workaround.

##  Version

* OCP 4.13 → OCP 4.17

## Preview

* [Restoring applications](https://81525--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.html)

## QE Review

* [x] QE has approved this change.

### CODE REVIEW

File: snippets/oadp-image-stream-tag-trigger.adoc
Comment (anarnold97) [inline snippets/oadp-image-stream-tag-trigger.adoc:9]:
```suggestion
During a File System Backup (FSB) restore operation, a `Deployment` resource referencing an `ImageStream` is not restored properly. Due to this, pod restoration `postHook` is not executed, and the volume content is not restored during. 
```

File: snippets/oadp-image-stream-tag-trigger.adoc
Comment (anarnold97) [inline snippets/oadp-image-stream-tag-trigger.adoc:11]:
```suggestion
This happens because, during the restore operation, an OpenShift controller updates the `spec.template.spec.containers[0].image` field in the `Deployment` resource with the `ImageStreamTag` hash. The update triggers the rollout of a new pod, affecting the FSB volume restoration and pod restoration `postHook` execution. For more information, see "Triggering updates on image stream changes".
```

Is `Triggering updates on image stream changes` meant to be a link?

File: snippets/oadp-image-stream-tag-trigger.adoc
Comment (anarnold97) [inline snippets/oadp-image-stream-tag-trigger.adoc:13]:
```suggestion
The workaround for this behavior can be a two-step restore process. In the first step, you perform a restore by excluding the `deploymentconfig` resources, for example `--exclude-resources=deployment.apps`, and in the second step, you perform a restore including these resources, for example, `--include-resources=deployment.apps`.
```

Comment (anarnold97) [inline snippets/oadp-image-stream-tag-trigger.adoc:13]:
```suggestion
The workaround for this behavior is a two-step restore process:

. Perform a restore by excluding the `deploymentconfig` resources, for example `--exclude-resources=deployment.apps`.
. Perform a restore including these resources, for example, `--include-resources=deployment.apps`.
```

You could even do it as an ordered list

**Comment by anarnold97** [comment]
@shdeshpa07 - just a few pedantic suggestions. Otherwise LGTM
Thanks

File: snippets/oadp-image-stream-tag-trigger.adoc
Comment (shdeshpa07) [reply snippets/oadp-image-stream-tag-trigger.adoc]: I have added the link as an xref in the 'Additional  resources'  section of the 'Restoring' assembly. This is per OCP guidelines. Also. we are including the snippet in RN too and the xref will probably fail to resolve.

File: snippets/oadp-image-stream-tag-trigger.adoc
Comment (shdeshpa07) [reply snippets/oadp-image-stream-tag-trigger.adoc]: Made it an ordered list. thanks.

**Comment by shdeshpa07** [comment]
@kaovilai  @amastbau  - Could you please review the PR? Thanks.

**Comment by amastbau** [comment]
/LGTM but please also get LGTM from tiger

File: snippets/oadp-image-stream-tag-trigger.adoc
Comment (kaovilai) [inline snippets/oadp-image-stream-tag-trigger.adoc:16]:
````suggestion
During a File System Backup (FSB) restore operation, a `Deployment` resource referencing an `ImageStream` is not restored properly. Restored Pod running File System Backup as well as `postHook` is killed prematurely due to a rollout triggerd by ImageStreamTag update.

This happens because, during the restore operation, OpenShift controller updates the `spec.template.spec.containers[0].image` field in the `Deployment` resource with an updated `ImageStreamTag` hash. The update triggers the rollout of a new pod, killing the pod velero is running File System Backup and post restore hook on. For more information about image stream trigger, see "Triggering updates on image stream changes".

The workaround for this behavior is a two-step restore process:

* First perform a restore excluding the `Deployment` resources, for example
```
velero restore create <RESTORE_NAME> \
  --from-backup <BACKUP_NAME> \
  --exclude-resources=deployment.apps
```
* Once first restore is successful, perform a second restore by including these resources, for example
```
velero restore create <RESTORE_NAME> \
  --from-backup <BACKUP_NAME> \
  --include-resources=deployment.apps
```
````

**Review by kaovilai — CHANGES_REQUESTED** [review]

**Review by kaovilai — APPROVED** [review]

File: snippets/oadp-image-stream-tag-trigger.adoc
Comment (shdeshpa07) [reply snippets/oadp-image-stream-tag-trigger.adoc]: Thanks for the review. I reworded a little to use 'terminating' instead of 'killing' :). Thanks.

**Comment by shdeshpa07** [comment]
@kaovilai  - I have incorporated your review comments. Please can you give a quick look and an LGTM. Thanks.

**Review by kaovilai — APPROVED** [review]

File: backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc
Comment (sheriff-rh) [inline backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc:24]: No period at the end, please.

Comment (sheriff-rh) [inline backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc:24]: No period at the end, please.

File: backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc
Comment (sheriff-rh) [inline backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc:16]: I think this "note" should be "important", since it relates to a workaround to restore a backup. Don't want users to miss this information!

File: snippets/oadp-image-stream-tag-trigger.adoc
Comment (sheriff-rh) [inline snippets/oadp-image-stream-tag-trigger.adoc:24]: Per IBM style guide: 

once adverb
Use only to mean one time. Do not use as a conjunction to mean after or when.

**Review by sheriff-rh — COMMENTED** [review]
I have some small suggestions. Also please squash your commits before putting it on merge review. Thanks!

File: backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc
Comment (shdeshpa07) [reply backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc]: Done. Thanks.

File: backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc
Comment (shdeshpa07) [reply backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc]: Done, thanks.

File: snippets/oadp-image-stream-tag-trigger.adoc
Comment (shdeshpa07) [reply snippets/oadp-image-stream-tag-trigger.adoc]: Done. Thanks.

**Review by opayne1 — COMMENTED** [review]
/lgtm

**Pull request merged**
